### PR TITLE
Moved get-jetpack-data to shared-extension-utils

### PR DESCRIPTION
**Pull request by mkm119**
Fixes #

#### Changes proposed in this Pull Request:

* Moved get-jetpack-data file from jetpack shared extensions folder to js-packages/shared-extension-utils. Also, updated the import references of getJetpackData in the Jetpack plugin codebase

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:

* Go to '..'
*

### CODE REVIEW

File: projects/packages/my-jetpack/changelog/master-merge-conflict-resolve [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Significance: patch
+Type: compat
```
Comment (pablinos) [inline projects/packages/my-jetpack/changelog/master-merge-conflict-resolve:2]: Where we've changed the use of `getJetpackData` that should probably be what we have in the changelog. It's maybe `changed` as a type?

File: projects/js-packages/shared-extension-utils/src/get-jetpack-data.js [hunk 2]
```diff
@@ -5,6 +5,11 @@ import { get } from 'lodash';
 
 export const JETPACK_DATA_PATH = 'Jetpack_Editor_Initial_State';
 
+/**
+ * Retrieves Jetpack editor state
+ *
+ * @returns {object|null}Object The Jetpack Editor State.
+ */
 export default function getJetpackData() {
 	return get( 'object' === typeof window ? window : null, [ JETPACK_DATA_PATH ], null );
```
Comment (pablinos) [inline projects/js-packages/shared-extension-utils/src/get-jetpack-data.js:14]: I wonder if we could use:

```suggestion
	return 'object' === typeof window ? window?.[ JETPACK_DATA_PATH ] ?? null : null;
```

and remove the lodash dependency

File: projects/packages/search/src/widgets/js/search-widget-admin.js [hunk 3]
```diff
@@ -1,11 +1,10 @@
 /* globals jetpack_search_filter_admin, jQuery, analytics */
 
-( function( $, args ) {
-	var defaultFilterCount = ( 'undefined' !== typeof args && args.defaultFilterCount ) ?
-		args.defaultFilterCount :
-		5; // Just in case we couldn't find the defaultFiltercount arg
+( function ( $, args ) {
```
Comment (pablinos) [inline projects/packages/search/src/widgets/js/search-widget-admin.js:3]: There are a lot of changes in this file to make the linter happy, but I can't see what we're changing. Is `getJetpackData` used in here? If not, then perhaps we can remove this and fix the code style issues in a separate PR.

**Review by pablinos — CHANGES_REQUESTED** [review]
Looking good so far, I think we just need to work out what of the `projects/packages/search/src/widgets/js/search-widget-admin.js` changes need to be part of this PR, if any.

File: projects/packages/search/src/widgets/js/search-widget-admin.js (hunk 3 above)
Comment (mkm119) [reply projects/packages/search/src/widgets/js/search-widget-admin.js:3]: So this file start showing up in the changes when I merged the master in my branch yesterday. It doesn't have any changes related to getJetpackData. Should I remove it from here? Not sure why it started showing in the modified file list in the first place :o

File: projects/packages/my-jetpack/changelog/master-merge-conflict-resolve (hunk 1 above)
Comment (mkm119) [reply projects/packages/my-jetpack/changelog/master-merge-conflict-resolve:2]: Master merge had some conflicts in this folder which I had to resolve. While pushing, it asked me to add a changelog file. It doesn't have anything to do with `getJetpackData`

File: projects/packages/search/src/widgets/js/search-widget-admin.js (hunk 3 above)
Comment (mkm119) [reply projects/packages/search/src/widgets/js/search-widget-admin.js:3]: Looks like I accepted the wrong change while resolving conflicts. [This file](https://github.com/Automattic/jetpack/blob/master/projects/packages/search/src/widgets/js/search-widget-admin.js) has the changes which I am trying to remove in this PR. I will reset this change in my PR and commit again. @pablinos

File: projects/js-packages/shared-extension-utils/src/get-jetpack-data.js (hunk 2 above)
Comment (mkm119) [reply projects/js-packages/shared-extension-utils/src/get-jetpack-data.js:14]: That should work. 

In fact, I believe '`object' === typeof window` check will not be required with the conditional` window?.[ JETPACK_DATA_PATH ]`

Making this change

Comment (pablinos) [reply projects/js-packages/shared-extension-utils/src/get-jetpack-data.js]: > I believe 'object' === typeof window check will not be required 

Yeah I thought that too, but in my tests if you do something like `undefVar?.property` you get an error that `undefVar` is undefined. If this is always called in the browser and not server side, then I think it's probably ok to remove the object type check.

File: projects/js-packages/shared-extension-utils/package.json [hunk 4]
```diff
@@ -17,6 +17,7 @@
 	},
 	"devDependencies": {
 		"jetpack-js-test-runner": "workspace:*",
+		"lodash": "4.17.21",
```
Comment (pablinos) [inline projects/js-packages/shared-extension-utils/package.json:20]: We can get rid of this now.

**Review by pablinos — COMMENTED** [review]
Looking at the tests that are failing it seems we've got to update references to `JETPACK_DATA_PATH` like this one:

```
import { JETPACK_DATA_PATH } from '../../../shared/get-jetpack-data';
```

File: projects/js-packages/shared-extension-utils/package.json (hunk 4 above)
Comment (mkm119) [reply projects/js-packages/shared-extension-utils/package.json:20]: We need this for other files like [get-jetpack-extension-availability](https://github.com/Automattic/jetpack/blob/b94c0bd9cafa1f05b4f837433765e01b87a2833d/projects/js-packages/shared-extension-utils/src/get-jetpack-extension-availability.js#L26). I will have to add it back again even if I remove it from this PR.

Comment (pablinos) [reply projects/js-packages/shared-extension-utils/package.json:20]: If it's just for `get` then we can remove the dependency there too. That instance could become something like:

```
const details = data?.available_blocks?.[name]?.details ?? [];
```

Comment (mkm119) [reply projects/js-packages/shared-extension-utils/package.json:20]: That sounds about right. Let me remove it for now. We can add it back if required.

File: composer.lock [hunk 5]
```diff
@@ -1022,12 +1022,12 @@
             "source": {
                 "type": "git",
                 "url": "https://github.com/WordPress/WordPress-Coding-Standards.git",
-                "reference": "73f3d69b3a200935b80f5fd265bfbee8d642d082"
+                "reference": "7b39722c38cbf2fe64c55be7154ee6f1807d304c"
```
Comment (pablinos) [inline composer.lock:1025]: This happened after running `tools/composer-update-monorepo.sh --root-reqs "."`, which was recommended as a fix to a failing check. I'm not sure if it's an issue to update this reference.

**Review by pablinos — APPROVED** [review]
This looks good to me. We'll have to work out how to deploy this to wpcom. I've added a couple of notes about things to help with the Crew review

**Review by jeherve — APPROVED** [review]

**Comment by jeherve** [comment]
r239736-wpcom

**Pull request merged**